norns/clocks.md

```diff
@@ -0,0 +1,211 @@
+---
+layout: default
+parent: scripting
+grand_parent: norns
+title: clocks
+nav_order: 3
+permalink: /norns/clocks/
+---
+
+# clocks
+
+the clock library provides a way to create timed function calls: loops, repetition, and delays. synchronization is possible and time base can come from a variety of sources.
+
+## commands
+
+- `id = clock.run( func )` &mdash; start a new clock with function _func_. returns _id_
+- `clock.cancel( id )` &mdash; cancel clock _id_ 
+- `clock.sleep( time )` &mdash; resume in _time_ seconds
+- `clock.sync( beats )` &mdash; resume in _beats_ according to tempo
+- `beats = clock.get_time_beats()` &mdash; returns current time in _beats_
+- `tempo = clock.get_tempo()` &mdash; returns current _tempo_
+- `beat_sec = clock.get_beat_sec()` &mdash; returns length of beat in seconds
+
+## callbacks
+
+these functions can be script defined and are called by system events:
+
+- `clock.transport.start` &mdash; called by transport start
+- `clock.transport.stop` &mdash; called by transport stop
+
+## use
+
+### simple delay
+
+`clock.run` starts a new clock with the function you provide:
+
+```
+function later()
+  clock.sleep(2)
+  print("now awake")
+end
+
+function init()
+  print("starting now")
+  clock.run(later)
+  print("done with init")
+end
```

Comment:
```suggestion
function init()
  print("starting now")
  clock.run(later)
  print("done with init")
end

function later()
  clock.sleep(2)
  print("now awake")
end
```

reads a bit more linearly to see `clock.run` (the first thing a reader is told to look for in this section) before `later()`.